crates/claims/crates/vc/src/syntax/non_empty_vec.rs

```diff
@@ -0,0 +1,90 @@
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq)]
+#[serde(try_from = "Vec<T>", into = "Vec<T>")]
+pub struct NonEmptyVec<T: Clone>(Vec<T>);
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("expected a non-empty array")]
+    Empty,
+}
+
+impl<T: Clone> NonEmptyVec<T> {
+    pub fn new(t: T) -> Self {
+        Self(vec![t])
+    }
+
+    pub fn maybe_new(v: Vec<T>) -> Option<Self> {
```

Comment:
The name convention would be `from_vec` or `try_from_vec` here, but it doesn't really matter.

Reply:
yes that's true, and I moved to a `Result`